packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -983,9 +1037,10 @@ class CupertinoTimerPicker extends StatefulWidget {
   /// positive integer factor of 60.
   CupertinoTimerPicker({
     this.mode = CupertinoTimerPickerMode.hms,
-    this.initialTimerDuration = Duration.zero,
+    this.initialTimerDuration = Duration.zero, // ignore: deprecated_member_use
     this.minuteInterval = 1,
     this.secondInterval = 1,
+    this.controller,
     @required this.onTimerDurationChanged,
   }) : assert(mode != null),
```

Comment:
We probably want to assert that not both initialTimerDuration and controller are set. Also assert that the controller's set value is sane (e.g. all the below asserts).

Reply:
I'm asserting the value is sane in the _desiredDurationChanged function.